guarddog/cli.py

```diff
@@ -55,6 +44,15 @@ def common_options(fn):
 
 
 def legacy_rules_options(fn):
+    ALL_RULES = reduce(
+        lambda a, b: a | b,
+        map(
+            lambda e: set([r.id for r in get_sourcecode_rules(e)])
```

Comment:
Here, I think you can also get away without the intermediate list: `set(r.id for r in get_sourcecode_rules(...))`

Reply:
Btw this is cool, I love maps and folds

Reply:
done in [aecc17c](https://github.com/DataDog/guarddog/pull/401/commits/aecc17cd62cc8b49845e267dec21b50f37ab0c11)